### DISCUSSION TITLE
lesson 7 - lottery

### DISCUSSION THREAD

**Discussion by prongs394** [smartcontractkit/full-blockchain-solidity-course-py#1743] [Q&A] [answered]
hi. i am learning the lottery lesson and im having problems getting the random number.
should i learn VRFConsumerBaseV2 or does RandomNumberConsumer which is in this course still work?
if RandomNumberConsumer what things should i change (ex rinkbey and kovan dont work anymore so how do i use goerli now? what should i change in the code)

**Comment by Niferu** [comment]
Hello @prongs394 
Using VRFConsumerBaseV1 should still work, and code remains the same. You have to just switch network to Goerli in your `brownie-config.yaml` file according with all its parameters.

For code errors please check repository and compare your code with it, or share your error with code here, so we can help maybe.

**Comment by Niferu** [comment, marked as answer]
Hi @prongs394
I have even tested it by myself for Goerli and everything is working perfectly. Let me know if you need any help with it.

**Reply by prongs394** [reply]
thanks for replying
i used this link :
https://docs.chain.link/docs/vrf/v1/examples/get-a-random-number/
but it didnt ask me for requestId when i deployed the contract on remix.
then i made a subscription in VRF chainlink and also gave it the contract address but when wanted to use the functions it said i dont have enough link.
can you help with this or do you have a link that explains this a little better?
i watched this video on youtube: https://www.youtube.com/watch?v=rdJ5d8j1RCg
but it was a little different (like it took a request id when deploying)

**Reply by Niferu** [reply]
Hi @prongs394
I thought you have problems deployig lottery via python scripts. First link you have posted is for VRF_v1 then you have passed procedures for VRF_v2. YouTube video you have posted is for VRF_v2 and mixing v1 and v2 won't work at all.

If you just want to get random number in remix for VRF_v1 do the following:
1. Open that remix code you sent first
2. In order to perform `getRandomNumber` button function you have to first send some `LINK` from your wallet to deployed contract address. Once transaction pass your deployed contract will have some `LINK` now to get you random number.
![image](https://user-images.githubusercontent.com/110537615/195654457-9de77452-bbc8-442f-a65f-08ec3e062900.png)
3. Click `getRandomNumber` button accept metaMask popup and wait until `randomResult` button throws you random number != than 0. (It can take pretty long, so be patient)

Hope this will help you ;)

**Reply by cromewar** [reply]
Nice Approach @Neftyr 
Also take in mind that the repos was already updated to work to with Goerli, so be sure to compare your code con the repo :)

**Reply by prongs394** [reply]
yeah i know the youtube vid is for vrf2
i checked vrf2 and it was different from the video
one question i have is why vrf1 does not take any requestId like it does in vrf2 video

so this is what i did:
1- in https://vrf.chain.link/ i click on create a subscription and with my metamask account i create a subscription with 5 link in it
2- deploy the contract 
3- copy what you have circled and paste it in vrf chainlink in add consumer
but now in remix above the getRandomNumber button it still says "Balance: 0 ETH" and then when i use getRandomButton i get:
Gas estimation failed - execution reverted: Not enough LINK - fill contract with faucet

i tried vrf_v2 once and it asked for requestId in constructor but in this one the constructor doesnt ask for one and i think thats why it doesnt work

**Reply by Niferu** [reply]
Hi @prongs394 
You are still mixing VRF_v1 with VRF_v2. Just follow my instruction point by point to get random number for VRF_v1, as I have mentioned earlier you got gas estimation error because you did not send any `LINK` tokens to your deployed contract, and that `Balance: 0 ETH` wont change at any point even if you will send `ETH` to your contract. If you want to use VRF_v1 you dont need adding any consumers or requestId's, you just need `LINK` tokens on your deployed contract. VRF_v2 is different than VRF_v1 and it requires adding contract to consumer list and also it requires requestId and it requires filling that contract address you added to consumer with LINK's.

**Reply by prongs394** [reply]
it worked thanks @Neftyr

**Reply by cromewar** [reply]
Good job the both of you!
